lib/charms/bind/v0/dns_record.py

```diff
@@ -238,15 +240,54 @@ class RequirerDomain(BaseModel):
     Attributes:
         domain: the domain name.
         username: username for authentication.
+        password: the password for authentication.
         password_id: secret password for authentication.
         uuid: UUID for this entry.
     """
 
     domain: str = Field(min_length=1)
     username: str
+    password: Optional[str] = Field(default=None, exclude=True)
     password_id: str
     uuid: UUID
 
+    def get_password(self, model: ops.Model) -> str:
+        """Retrieve the password corresponding to the password_id.
+
+        Args:
+            model: the Juju model.
+
+        Returns:
+            the plain password.
+        """
+        secret = model.get_secret(id=self.password_id)
+        password = secret.get_content().get("domain-password")
+        # The content is always present in the secret.
+        assert password
+        return password
+
+    def set_password_id(self, model: ops.Model, relation: ops.Relation) -> None:
+        """Store the password as a Juju secret.
+
+        Args:
+            model: the Juju model
+            relation: relation to grant access to the secrets to.
+        """
+        if not self.password:
+            return
+        secret = model.get_secret(label=f"{self.domain}")
```

Comment:
I'm worried about the scalability of having 1 secret per domain... that could lead to manipulate hundreds of secrets. We should probably check with the Juju team for any recommendation here.
We might want to store a more complex object in a single secret per relation

Reply:
Tbh, I wouldn't expect that many secrets per relation. What use cases do you have in mind?

Reply:
It looks like content-cache has, in some deployed cases, 50 sites configured, so we could potentially be requesting 50 certificates.

Reply:
We've been discussing this topic with Arturo, and we decided to adapt the relation interface to limit the number of required secrets. That will be managed in a following PR